src/libasr/pass/pass_array_by_data.cpp

```diff
@@ -75,7 +101,7 @@ class PassArrayByDataProcedureVisitor : public PassUtils::PassVisitor<PassArrayB
                     std::to_string(tmp_expr->type));
             }
             std::string x_sym_name = std::string(ASRUtils::symbol_name(x_sym));
-            if( current_proc_scope->get_symbol(x_sym_name) != x_sym ) {
+            if( current_proc_scope->get_symbol(x_sym_name) && current_proc_scope->get_symbol(x_sym_name) != x_sym ) {
```

Comment:
Make it the following,

```suggestion
            if( current_proc_scope->resolve_symbol(x_sym_name) != x_sym ) {
```

Reply:
@czgdp1807 it seems this fix did not work. Minipack compilation still fails.

Reply:
Minpack compilation was related to https://github.com/lfortran/lfortran/blob/deb490e679a387ca9499b9211747c8a6e654b8d3/integration_tests/minpack_02.f90 which I fixed in https://github.com/lfortran/lfortran/pull/1392/commits/326fccb17134c2c12f9ce5d413bc53bbaf0f8cb7.